seed/static/seed/js/services/uploader_service.js

```diff
@@ -126,6 +126,56 @@ angular.module('BE.seed.service.uploader', []).factory('uploader_service', [
       }, failure_fn);
     };
 
+    uploader_factory.check_progress_loop_main_sub = function (progress_argument, success_fn, failure_fn, sub_progress_argument=null) {
+      const {progress_key} = progress_argument
+      const sub_progress_key = sub_progress_argument ? sub_progress_argument.progress_key : null 
+
+      let progress_list = [uploader_factory.check_progress(progress_key)]
+      sub_progress_argument && progress_list.push(uploader_factory.check_progress(sub_progress_key))
+
+      Promise.all(progress_list).then((values) => {check_and_update_progress(values)})
+      
+      function check_and_update_progress(data) {
+        $timeout(function () {
+          progress_argument = update_progress_bar_obj(data[0], progress_argument)
+          if (data[0].progress < 100) {
+            data.length > 1 ? (
+              sub_progress_argument = update_progress_bar_obj(data[1], sub_progress_argument),
+              uploader_factory.check_progress_loop_main_sub(progress_argument, success_fn, failure_fn, sub_progress_argument)
+              ) :
+              uploader_factory.check_progress_loop_main_sub(progress_argument, success_fn, failure_fn)
+          } else {
+            success_fn(data[0]);
+          }
+          
+        }, 750);
+      };
+
+      function update_progress_bar_obj(data, progress_arg) {
```

Comment:
Yoo this is nice, could you reuse it in the existing `check_progress_loop` above?